### DISCUSSION TITLE
Append text field to slider

### DISCUSSION THREAD

**Discussion by EZoni** [Kitware/trame#596] [Q&A] [answered]
Hi again,

The following code creates a slider with an appended text field box, which is synchronized with the values on the slider:
```py
from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout
from trame.widgets import vuetify

server = get_server(client_type="vue2")
state, ctrl = server.state, server.controller

state.input = 0.5

with SinglePageLayout(server) as layout:
    with layout.content:
        with vuetify.VContainer():
            with vuetify.VRow():
                with vuetify.VCol():
                    with vuetify.VCard(style="width: 500px"):
                        with vuetify.VCardText():
                            with vuetify.VSlider(
                                label="input",
                                min=0.,
                                max=1.,
                                step=0.1,
                                thumb_label=True,
                                v_model=("input",),
                            ):
                                with vuetify.Template(
                                    v_slot_append=True,
                                    __properties=[("v_slot_append", "v-slot:append")],
                                ):
                                    vuetify.VTextField(
                                        v_model=("input",),
                                    )

if __name__ == "__main__":
    server.start()
```

The code works, but the slider and the text field box are displayed as follows:
![Screenshot from 2024-09-24 16-35-55](https://github.com/user-attachments/assets/3a1c954f-4f30-4e33-90ad-458cf38466db)

Instead, I would like to display them as follows:
![Screenshot from 2024-09-24 16-35-30](https://github.com/user-attachments/assets/a82a40e5-8b51-402f-b04c-de3940cd2242)

I did not find in the docs how to get the alignment between the text field box and the slider right.

The second figure, with the display that I would like to reproduce, is from the examples at https://vuetifyjs.com/en/components/sliders/. In particular, its source code, copied from that website, is
```vue
<template>
  <v-slider
    v-model="slider"
    :max="max"
    :min="min"
    class="align-center"
    hide-details
  >
    <template v-slot:append>
      <v-text-field
        v-model="slider"
        density="compact"
        style="width: 70px"
        type="number"
        hide-details
        single-line
      ></v-text-field>
    </template>
  </v-slider>
</template>
<script>
  export default {
    data () {
      return {
        min: -50,
        max: 90,
        slider: 40,
      }
    },
  }
</script>
```

I don't know what I missed in the translation of this code to my Python code that is responsible for the bad alignment in my GUI. 

Do you know how to fix my Python code so that the text field box is better aligned with the slider?

Thanks in advance!

**Comment by jourdain** [comment]
You are missing some property on the text field in your python

```
density="compact",
style="width: 70px",
type="number",
hide_details=True.
single_line=True,
```

**Comment by EZoni** [comment]
Thanks! 

Adding these properties does make it look a bit more similar to the original version, but the vertical alignment seems to be still off:
![Screenshot from 2024-09-24 17-08-31](https://github.com/user-attachments/assets/392543c1-67de-4ae3-b251-224b3196139f)

Is it maybe due to the property `class="align-center"` of the slider? 
If so, what would be the corresponding parameter name in Python?

**Comment by jourdain** [comment]
Your slider don't have the same props either... Missing `hide_details=True`...

**Reply by jourdain** [reply]
Since `class` is reserved in python, it has been mapped to `classes=""`

**Reply by jourdain** [reply]
See https://trame.readthedocs.io/en/latest/core.widget.html

**Comment by EZoni** [comment, marked as answer]
Thanks! 

The argument that seems to be more decisive in fixing the vertical alignemnt is `classes="align-center"`, for the slider.

Here's the final working code:
```py
import numpy as np

from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout
from trame.widgets import vuetify

server = get_server(client_type="vue2")
state, ctrl = server.state, server.controller

state.input = np.float64(5e-2)

with SinglePageLayout(server) as layout:
    with layout.content:
        with vuetify.VContainer():
            with vuetify.VRow():
                with vuetify.VCol():
                    with vuetify.VCard(style="width: 500px"):
                        with vuetify.VCardText():
                            with vuetify.VSlider(
                                label="input",
                                min=np.float64(0e-1),
                                max=np.float64(1e-1),
                                step=np.float64(1e-2),
                                v_model=("input",),
                                hide_details=True,
                                classes="align-center",
                            ):
                                with vuetify.Template(
                                    v_slot_append=True,
                                    __properties=[("v_slot_append", "v-slot:append")],
                                ):
                                    vuetify.VTextField(
                                        v_model=("input",),
                                        density="compact",
                                        style="width: 70px",
                                        hide_details=True,
                                        single_line=True,
                                    )

if __name__ == "__main__":
    server.start()
```

**Comment by EZoni** [comment]
Just as a follow-up question, I learned that I had to write something like 
```py
with vuetify.Template(
    v_slot_append=True,
    __properties=[("v_slot_append", "v-slot:append")],
):
    ...
```
in order to _append_ the text field box to the slider, by reading the example in the discussion #558.

If I didn't know about #558, where in [docs](https://trame.readthedocs.io/en/latest/index.html) should I look in order to understand that this
```vue
<template v-slot:append>
    ...
</template>
```
can be translated to Python via
```py
with vuetify.Template(
    v_slot_append=True,
    __properties=[("v_slot_append", "v-slot:append")],
):
    ...
```
just so I make sure I understand how to properly read the docs next time?

**Reply by jourdain** [reply]
In one of the slides available https://kitware.github.io/trame/guide/intro/course.html
